docs/docs/reference-react-component.md

```diff
@@ -0,0 +1,170 @@
+
+### `.displayName`
+
+The `displayName` string is used in debugging messages. JSX sets this value automatically; see [JSX in Depth](/react/docs/jsx-in-depth.html).
+
+### `.forceUpdate(callback)`
+
+TODO
+
+### `.props`
+
+`this.props` contains the props that were defined by the caller of this component. See [Components and Props](/react/docs/components-and-props.html) for an introduction to props.
+
+In particular, `this.props.children` is a special prop, typically defined by the child tags in the JSX expression rather than in the tag itself.
+
+### `.propTypes`
+
+`propTypes` can be defined as a property on the component class itself, to define what types the props should be. It should be a map from prop names to types as defined in `React.PropTypes`. In development mode, when an invalid value is provided for a prop, a warning will be shown in the JavaScript console. In production mode, `propTypes` checks are skipped for efficiency.
+
+For example, this code ensures that the `color` prop is a string:
+
+```js
+class CustomButton extends React.Component {
+  // ...
+}
+
+CustomButton.propTypes = {
+  name: React.PropTypes.string
+};
+```
+
+### `.render()`
+
+The `render()` method is required.
+
+When called, it should examine `this.props` and `this.state` and return a single React element. This element can be either a representation of a native DOM component, such as `<div />`, or another composite component that you've defined yourself.
+
+You can also return `null` or `false` to indicate that you don't want anything rendered. React will just render a comment tag. When returning `null` or `false`, `ReactDOM.findDOMNode(this)` will return `null`.
+
+The `render()` function should be pure, meaning that it does not modify component state, it returns the same result each time it's invoked, and it does not directly interact with the browser. If you need to interact with the browser, perform your work in `componentDidMount()` or the other lifecycle methods instead. Keeping `render()` pure makes components easier to think about.
+
+### `.setState(nextState, callback)`
+
+```javascript
+void setState(
+  function|object nextState,
+  [function callback]
+)
+```
+
+Performs a shallow merge of nextState into current state. This is the primary method you use to trigger UI updates from event handlers and server request callbacks.
+
+The first argument can be an object (containing zero or more keys to update) or a function (of state and props) that returns an object containing keys to update.
+
+Here is the simple object usage:
+
+```javascript
+this.setState({mykey: 'my new value'});
+```
+
+It's also possible to pass a function with the signature `function(state, props) => newState`. This can be useful in some cases when you want to enqueue an atomic update that consults the previous value of state and props before setting any values. For instance, suppose we wanted to increment a value in state:
```

Comment:
The props object is, more precisely, the value of the props at the time the update is applied. Calling it the "current props" could imply that it's the value at the time `setState` is called.